src/ethereum/osaka/bal_builder.py

```diff
@@ -0,0 +1,152 @@
+"""
+Block Access List Builder for EIP-7928
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This module implements the BAL builder that tracks all account and storage
+accesses during block execution and constructs the final BlockAccessList.
```

Comment:
New contributions should use markdown-formatted docstrings. See [`blocks.py`](https://github.com/ethereum/execution-specs/blob/forks/osaka/src/ethereum/osaka/blocks.py) for a recent example.